src/libasr/asr_verify.cpp

```diff
@@ -63,7 +63,7 @@ class VerifyVisitor : public BaseWalkVisitor<VerifyVisitor>
             diagnostics.message_label("ASR verify: " + error_msg,
                 {loc}, "failed here",
                 diag::Level::Error, diag::Stage::ASRVerify);
-            throw VerifyAbort();
+            LFORTRAN_ASSERT(false);
```

Comment:
I think you have to abort here, otherwise you won't get the error message?

Reply:
Well, I added this line for my own conviction to make the stacktrace include the code path which causes the ASR verify to fail. Otherwise it always stops at call to ASR verify and doesn’t include what caused the failure.

Reply:
It should work with `--show-stacktraces`, as it records the stacktrace to the failure when the exception was thrown. Does that not work?

Reply:
No it doesn't. See example below,

```bash
Internal Compiler Error: Unhandled exception
Traceback (most recent call last):
  File "/Users/czgdp1807/lpython_project/lpython/src/bin/lpython.cpp", line 1492
    return emit_llvm(arg_file, runtime_library_dir, lpython_pass_manager, compiler_options);
  File "/Users/czgdp1807/lpython_project/lpython/src/bin/lpython.cpp", line 547
    res = fe.get_llvm3(*asr, pass_manager, diagnostics, infile);
  File "/Users/czgdp1807/lpython_project/lpython/src/lpython/python_evaluator.cpp", line 58
    run_fn, infile);
  File "/Users/czgdp1807/lpython_project/lpython/src/libasr/codegen/asr_to_llvm.cpp", line 6448
    pass_manager.apply_passes(al, &asr, pass_options, diagnostics);
  File "/Users/czgdp1807/lpython_project/lpython/src/libasr/pass/pass_manager.h", line 196
    _apply_passes(al, asr, _passes, pass_options, diagnostics);
  File "/Users/czgdp1807/lpython_project/lpython/src/libasr/pass/pass_manager.h", line 99
    LFORTRAN_ASSERT(LFortran::asr_verify(*asr, true, diagnostics));
AssertFailed: LFortran::asr_verify(*asr, true, diagnostics)
```
Ends there, we don't have the paths of `asr_verify.cpp` in the stacktrace. I think this is because `VerifyAbort` is getting caught at the following line,

https://github.com/lcompilers/lpython/blob/93cc7c2058da1c902a032a2a8338697625dada7d/src/libasr/asr_verify.cpp#L709-L714

Reply:
I think the issue is that the `verify()` call itself is wrapped in an assert:
```
LFORTRAN_ASSERT(LFortran::asr_verify(*asr, true, diagnostics));
```
I have reworked this at a few places, but we need to fix it everywhere. See this in `asr_utils`:
```
#if defined(WITH_LFORTRAN_ASSERT)
        diag::Diagnostics diagnostics;
        if (!asr_verify(*tu, true, diagnostics)) {
            std::cerr << diagnostics.render2();
            throw LCompilersException("Verify failed");
        };
#endif
```
When you do this, it will print the stacktrace for verify failures.